### ISSUE TITLE
Bump dependencies for RISC-V support

### ISSUE THREAD

**Issue by maleadt** [zed-industries/zed#18891]
### Check for existing issues

- [X] Completed

### Describe the feature

The manual currently states:

> You will need to build from source for:
>
>  - architectures other than 64-bit Intel or 64-bit ARM (for example a 32-bit or RISC-V machine)

However, zed currently fails to build on RISC-V because of an (indirect) dependency on ring 0.16, which does not support RISC-V. Version 0.17 of the crate does, but updating that dependency is not straightforward to me (not being a Rust developer). The compatibility seems restricted by rustls and async-tungstenite, but bumping those dependencies to resp. 0.21.8 and 0.25 (which are the minimum versions that are compatible with ring 0.17) results in the introduction of http 1.1.0, which is incompatible with version 0.2.12 of the http crate that's used by other dependencies.

Would it be possible to upgrade dependencies in order to unblock RISC-V support? ring 0.16.20 is 3 years old, rustls 0.20.3 is over 2 years old, so it seems that upgrading these dependencies is a good idea anyway.

**Comment by notpeter** [comment]
There is active work in this area, not specifically to support RISC-V, but to using rustls and improving our http client.  See:
- https://github.com/zed-industries/zed/pull/19104

And the [recent Cargo.lock changes](https://github.com/zed-industries/zed/commits/main/Cargo.lock) for more.

**Comment by ifsheldon** [comment]
I was going to file a new issue but I saw this. I fixed this by updating two dependencies in `Cargo.toml` of the whole workspace:
```diff
- async-tungstenite = "0.24"
+ async-tungstenite = "0.28"

- rustls = "0.20.3"
+ rustls = "0.21.12"
```

I have successfully built `remote-server` which depends on `client` -> `async-tungstenite` -> `ring`.

I can open a PR of just these two lines in no time. But `Cargo.lock` should also be update.

Linking this to #19697 as I can build a RISC-V remote server.

**Comment by ConradIrwin** [comment]
Great! Assuming that doesn’t break other builds happy to merge a PR

**Comment by rjzak** [comment]
Building on RISC-V 64-bit currently fails due to a binary dependency on `webrtc-sys`.

```
error: failed to run custom build command for `webrtc-sys v0.3.5 (https://github.com/zed-industries/livekit-rust-sdks?rev=060964da10574cd9bf06463a53bf6e0769c5c45e#060964da)`

Caused by:
  process didn't exit successfully: `/home/rjzak/bin/zed/target/release/build/webrtc-sys-1114650fffbbaa99/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=LK_DEBUG_WEBRTC
  cargo:rerun-if-env-changed=LK_CUSTOM_WEBRTC
  cargo:CXXBRIDGE_PREFIX=webrtc-sys
  cargo:CXXBRIDGE_DIR0=/home/rjzak/bin/zed/target/release/build/webrtc-sys-c7a0990b1b77d9c0/out/cxxbridge/include
  cargo:CXXBRIDGE_DIR1=/home/rjzak/bin/zed/target/release/build/webrtc-sys-c7a0990b1b77d9c0/out/cxxbridge/crate

  --- stderr

  CXX include path:
    /home/rjzak/bin/zed/target/release/build/webrtc-sys-c7a0990b1b77d9c0/out/cxxbridge/include
    /home/rjzak/bin/zed/target/release/build/webrtc-sys-c7a0990b1b77d9c0/out/cxxbridge/crate
  thread 'main' panicked at /home/rjzak/.cargo/git/checkouts/livekit-rust-sdks-171a3090be725d50/060964d/webrtc-sys/build.rs:85:45:
  called `Result::unwrap()` on an `Err` value: "failed to download webrtc from https://github.com/livekit/client-sdk-rust/releases/download/webrtc-dac8015-6/webrtc-linux-riscv64-release.zip: 404 Not Found"
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

This reminds me of https://github.com/serde-rs/serde/issues/2538. Why download binaries to be compiled in?
